pkg/types/config.go

```diff
@@ -1,20 +1,26 @@
 package types
 
 type PlatformConfig struct {
-	Brand                 Brand             `yaml:"brand,omitempty"`
-	Version               string            `yaml:"version"`
-	Velero                *Velero           `yaml:"velero,omitempty"`
-	CA                    *CA               `yaml:"ca"`
-	Calico                Calico            `yaml:"calico,omitempty"`
-	CertManager           CertManager       `yaml:"certmanager,omitempty"`
-	Consul                string            `yaml:"consul"`
-	Dashboard             Dashboard         `yaml:"dashboard,omitempty"`
-	Datacenter            string            `yaml:"datacenter"`
-	DNS                   *DynamicDNS       `yaml:"dns,omitempty"`
-	DockerRegistry        string            `yaml:"dockerRegistry,omitempty"`
-	Domain                string            `yaml:"domain"`
-	EventRouter           *Enabled          `yaml:"eventRouter,omitempty"`
-	Harbor                *Harbor           `yaml:"harbor,omitempty"`
+	Brand       Brand       `yaml:"brand,omitempty"`
+	Version     string      `yaml:"version"`
+	Velero      *Velero     `yaml:"velero,omitempty"`
+	CA          *CA         `yaml:"ca"`
+	Calico      Calico      `yaml:"calico,omitempty"`
+	CertManager CertManager `yaml:"certmanager,omitempty"`
+	// Consul specifies the endpoint for externally hosted consul cluster
+	// NOTE: a working consol config required to verify
+	//       that primary master is available for a vSphere cluster.
+	Consul         string      `yaml:"consul"`
+	Dashboard      Dashboard   `yaml:"dashboard,omitempty"`
+	Datacenter     string      `yaml:"datacenter"`
+	DNS            *DynamicDNS `yaml:"dns,omitempty"`
+	DockerRegistry string      `yaml:"dockerRegistry,omitempty"`
+	// Domain specifies the domain that cluster will be available at
+	// NOTE: domain must be supplied for vSphere clusters.
```

Comment:
It would need to be specified for kind clusters as well, otherwise, a lot of functionality will not work/deploy